lib/src/value/calculation.dart

```diff
@@ -158,6 +343,230 @@ class SassCalculation extends Value {
+  }
+
+  /// Creates a `round()` calculation with the given [strategyOrNumber], [numberOrStep], and [step].
+  /// Strategy must be either nearest, up, down or to-zero.
+  ///
+  /// Number and step must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value round(Object strategyOrNumber,
+      [Object? numberOrStep, Object? step]) {
+    switch ((
+      _simplify(strategyOrNumber),
+      numberOrStep.andThen(_simplify),
+      step.andThen(_simplify)
+    )) {
+      case (SassNumber number, null, null):
+        return _matchUnits(number.value.round().toDouble(), number);
+
+      case (SassNumber number, SassNumber step, null):
+        _verifyCompatibleNumbers([number, step]);
+        return _roundWithStep('nearest', number, step);
+
+      case (
+          SassString(text: 'nearest' || 'up' || 'down' || 'to-zero') &&
+              var strategy,
+          SassNumber number,
+          SassNumber step
+        ):
+        _verifyCompatibleNumbers([number, step]);
+        return _roundWithStep(strategy.text, number, step);
+
+      case (
+          SassString(text: 'nearest' || 'up' || 'down' || 'to-zero'),
+          SassNumber() || SassString(isVar: true),
+          null
+        ):
+        throw SassScriptException("If strategy is not null, step is required.");
+
+      case (
+          SassString(text: 'nearest' || 'up' || 'down' || 'to-zero'),
+          null,
+          null
+        ):
+        throw SassScriptException(
+            "Number to round and step arguments are required.");
+
+      case (CalculationOperation number, null, null):
+        throw SassScriptException(
+            "Single argument $number expected to be simplifiable.");
+
+      case (SassString() && var strategy, null, null):
+        return SassCalculation._("round", [strategy]);
+
+      case (
+          (SassString() || CalculationOperation() || SassNumber()) &&
+              var strategy,
+          Object number,
+          null
+        ):
+        return SassCalculation._("round", [strategy, number]);
+
+      case (
+          (SassString(text: 'nearest' || 'up' || 'down' || 'to-zero') ||
+                  SassString(isVar: true)) &&
+              var strategy,
+          Object number,
+          Object step
+        ):
+        return SassCalculation._("round", [strategy, number, step]);
+
+      case (_, _?, _?):
+        throw SassScriptException(
+            "$strategyOrNumber must be either nearest, up, down or to-zero.");
+
+      case (_, _, _):
```

Comment:
By this point, you _should have_ verified all of the possible reasonable combinations of values. We can make use of Dart's pattern matching logic to verify that statically for us, but only if we don't add a catch-all here. Instead, we could write a final case for `(_, null, _?)` (which won't ever be generated by Sass code) and use Dart's static analysis to ensure we handle all the other cases above.

Reply:
Added a comment to ignore analyzer warning and follow-up issue: https://github.com/dart-lang/sdk/issues/52908